### ISSUE TITLE
Claim.getIndexId() fails with Error: fromBytes error: byte array incorrect length

### ISSUE THREAD

**Issue by nedgar** [iden3/js-iden3-core#16]
The [code](https://github.com/iden3/js-iden3-core/blob/main/src/claim.ts#L366) for `Claim.getIndexId()` passes the 32 byte value from slot 1 directly to `Id.fromBytes()`, which expects 31 bytes.

For example:

```ts
let id = core.Id.fromString("ww9xPmW6U3k6XatpaPPWbqPgUqW2Ct6AhsGNuYGm2")
let claim = new core.Claim()
claim.setIndexId(id)
claim.getIndexId()
==>
.../node_modules/@iden3/js-iden3-core/dist/cjs/id.js:53
            throw new Error('fromBytes error: byte array incorrect length');
            ^

Error: fromBytes error: byte array incorrect length
    at Id.fromBytes (.../node_modules/@iden3/js-iden3-core/dist/cjs/id.js:53:19)
    at Claim.getIndexId (.../node_modules/@iden3/js-iden3-core/dist/cjs/claim.js:329:24)
...
```

This works OK:
```ts
core.Id.fromBytes(claim.index[1].bytes.slice(0, -1)).string()
==>
ww9xPmW6U3k6XatpaPPWbqPgUqW2Ct6AhsGNuYGm2
```

**Comment by nedgar** [comment]
Also, a suggestion: it would be nice if there was a `getSubjectId()` method, which would interpret the subject flags and return the index ID, value ID, or `null` accordingly.

**Comment by nedgar** [comment]
Looks like `getValueId()` has the same problem.

**Comment by vmidyllic** [comment]
fixed in #17,  check v1.0.0-beta.1 version of the package
